foundry/collections/entityCollection.d.ts

```diff
@@ -61,62 +74,48 @@ declare abstract class EntityCollection<T extends Entity = Entity> extends Colle
   get object(): () => T;
 
   /**
-   * Initialize the Map object and all its contained entities
-   * @internal
+   * Add a new Entity to the EntityCollection, asserting that they are of the
+   * correct type.
+   * @param entity - The entity instance to add to the collection
    */
-  protected _initialize(data: T['data'][]): void;
+  insert(entity: T): void;
 
   /**
-   * Apply data transformations when importing an Entity from a Compendium pack
-   * @param data - The original Compendium entry data
-   * @returns The processed data ready for Entity creation
+   * Remove an Entity from the EntityCollection by its ID.
+   * @param id - The entity ID which should be removed
    */
-  fromCompendium(data: object): object;
+  remove(id: string): void;
 
   /**
    * Import an Entity from a compendium collection, adding it to the current
    * World.
    * @param collection - The name of the pack from which to import
-   * @param entryId - The ID of the compendium entry to import
+   * @param entryId    - The ID of the compendium entry to import
    * @param updateData - Optional additional data used to modify the imported
    *                     Entity before it is created
    *                     (default: `{}`)
-   * @param options - Optional arguments passed to the Entity.create method
-   *                  (default: `{}`)
+   * @param options    - Optional arguments passed to the Entity.create method
+   *                     (default: `{}`)
    * @returns A Promise containing the imported Entity
    */
   importFromCollection(collection: string, entryId: string, updateData?: object, options?: object): Promise<Entity>;
 
   /**
-   * Add a new Entity to the EntityCollection, asserting that they are of the
-   * correct type.
-   * @param entity - The entity instance to add to the collection
-   */
-  insert(entity: T): void;
-
-  /**
-   * Remove an Entity from the EntityCollection by its ID.
-   * @param id - The entity ID which should be removed
-   */
-  remove(id: string): void;
-
-  /**
-   * Render any Applications associated with this EntityCollection
-   * @param force - (default: `false`)
-   * @returns A reference to the rendered EntityCollection
-   * @see {@link Application.render}
+   * Apply data transformations when importing an Entity from a Compendium pack
+   * @param data - The original Compendium entry data
+   * @returns The processed data ready for Entity creation
    */
-  render(force?: boolean, options?: Application.RenderOptions): this;
+  fromCompendium(data: object): object;
```

Comment:
This should take and return `DeepPartial<T['data']>` I believe.

Reply:
✔ and importFromCollection got better types as well good catch.